requirements.txt

```diff
@@ -0,0 +1,154 @@
+ipython==8.23.0
```

Comment:
This file should be deleted, the dependencies are in pyproject.toml

Reply:
Here too, I'm new to python. My understanding so far, the .toml file contains the runtime or "production" configuration, while requirements.txt specifies the development setup. 

**Here is my reasoning**

That is, .toml should contain neither pytest nor hypothesis, but requirements.txt should contain both.

Second, as developer I would like to have two things:

1. exact dependencies that are proofed to work and where all tests pass successfully (freezed requirements.txt)
2. loose dependencies to receive latest updates of libraries and develop with most recent versions (dev-requirements.txt)

Reply:
honestly if there is an update to a library that causes an issue, we should want to find out in the development cycle. freezing the requirements like so silences the issue until users find it

Reply:
Well, I see your point. Still, I couldn't work or contribute for weeks because the tests failed.
From my perspective, the idea of the master branch in Git is that there is a common, working base. This why I suggested a dev_requirements and a freezed_requirements. This solution would take both views into account.

Reply:
how were the tests failing? and how did you end up fixing them?

Reply:
easy to reproduce: few weeks ago I fetched the main branch: commit 98c7b5884536cc5c333424a2d257b0443eb1c319

Then I did:

> source .venv/bin/activate
 pip install git+https://github.com/vyperlang/titanoboa
 pip install titanoboa==0.1.9

The last command is necessary since you published a newer version of titanoboa meanwhile.
Finally I executed
> make test

Result: 23 failed, 168 passed, 2 warnings, 71 errors in 19.26s

If I use the requirements.txt as specified in this PR all tests pass.